bin/ch-test

```diff
@@ -297,40 +305,102 @@ test_run () {
     set +e
 }
 
-test_make_auto () {
-    local img
-
-    printf "finding images compatible with test settings ...\n"
-    for i in $images $examples; do
-        if (test_make_check_image "$i"); then
-            echo "found: $i"
-            targets+=( "$i" )
-        fi
-    done
-
-    printf '\n'
+test_make () {
+    local bat_file
+    local tag
+    printf "finding artifacts compatible with %s test settings ...\n" "$phase"
     case $phase in
     build)
+        for i in $images $examples; do
+           if (test_make_check_image "$i"); then
+               echo "found: $i"
+               build_targets+=( "$i" )
+           fi
+        done
+        printf '\n'
         printf 'generate build_auto.bats ...\n'
-        "$make_auto" build "${targets[@]}" > "${TMP_}/build_auto.bats"
+        test_make_auto "$phase" "${build_targets[@]}" > "${TMP_}/build_auto.bats"
         printf 'ok\n\n'
         ;;
     run)
+        for i in "${CH_TEST_TARDIR}"/*{.tar.gz,.sqfs}; do
+            if [[ -e $i ]]; then
+                artifact=${i##*/}
```

Comment:
I'd call it something related to "packed image" rather than "artifact" — more specific.